### PR TITLE
remove caching of macro names and values in glslc

### PR DESCRIPTION
**Pull request by Qining**

### CODE REVIEW

File: glslc/src/main.cc [hunk 1]
```diff
@@ -212,23 +205,15 @@ int main(int argc, char** argv) {
...
-        // string_piece one. The original implementation doesn't append '\0' but
-        // works fine with libshaderc_util because it has 'end_' property. But
-        // libshaderc api doesn't know that so the macro 'name' will include the
-        // 'value' if we don't insert '\0'.
-
-        macro_names.emplace_back(name_piece.data(), name_length);
-        macro_values.emplace_back(
-            equal_sign_loc == shaderc_util::string_piece::npos ||
+        const string_piece value_piece =
+            equal_sign_loc == string_piece::npos ||
```
Comment (dneto0) [inline glslc/src/main.cc]: Please put parentheses around the expression before the ?
Makes it clearer and idiot-proof.

**Comment by dneto0** [comment]
-1 for a nit on parenthesizing the expression.

File: glslc/src/main.cc (hunk 1 above)
Comment (Qining) [inline glslc/src/main.cc]: Yes, I should have it in parentheses before. Now it looks better.

**Comment by dneto0** [comment]
Can you explain the Windows Release failure in AppVeyor?
Otherwise looks ok.

**Comment by Qining** [comment]
I added the parentheses at a wrong place, changed the '=' logical and collapsed the whole macro_value string_piece.

Should have already been fixed.

**Comment by dneto0** [comment]
+2

**Comment by dneto0** [comment]
Rebased and pushed into master as e145d34a0

**Pull request closed without merging**